service-base/src/main/java/org/eclipse/hono/service/registration/BaseRegistrationService.java

```diff
@@ -304,6 +322,50 @@ public void assertRegistration(
         }).setHandler(resultHandler);
     }
 
+    private Future<RegistrationResult> updateDeviceLastViaIfNeeded(final String tenantId, final String deviceId,
```

Comment:
Why not simply return `Future<Void>`? Why is this method required to create the result of the overall operation it is invoked as part of (and which this operation doesn't need to know anything about)?

Reply:
It's now changed to return `Future<Void>`.